tests/test_desktop_integration.cpp

```diff
@@ -0,0 +1,185 @@
+// library headers
+#include <glib.h>
+#include <gtest/gtest.h>
+#include <glib/gstdio.h>
+#include <appimage/appimage.h>
+#include <fcntl.h>
+extern "C" {
+#include <unistd.h>
+}
+
+#include "desktop_integration.h"
+#include "file_management_utils.hpp"
+
+class DesktopIntegrationTests : public ::testing::Test {
+protected:
+    std::string appdir_path;
+    std::string user_dir_path;
+    char* appimage_path;
+
+    virtual void SetUp() {
+        appimage_path = g_strjoin("/", TEST_DATA_DIR, "Echo-x86_64.AppImage", NULL);
+
+        appdir_path = createTempDir("libappimage-di-appdir");
+        user_dir_path = createTempDir("libappimage-di-user-dir");
+
+        ASSERT_FALSE(appdir_path.empty());
+        ASSERT_FALSE(user_dir_path.empty());
+    }
+
+    virtual void TearDown() {
+        removeDirRecursively(appdir_path);
```

Comment:
Teste here: https://github.com/AppImage/libappimage/blob/06bc4d3cebdde4557ab4459c079c58759d7488c5/tests/test_desktop_integration.cpp#L65
I guess that it could be used in another place and mixing it with the test code will limit that.

Reply:
Tested in the unit tests, I mean.